contrib/gitready.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash -eu
+
+git_commit() {
+    if ! git diff-index --quiet HEAD; then
+        git add -p
+        # returns 1 if cached index is not empty
+        git diff --cached --exit-code || git commit -m "$title"
+        git reset --hard
```

Comment:
why is it needed? Don't we expect that there are no any uncommited changes at this stage?

Reply:
while handling `git add -p` it might be that we did not stage all buffers, hence the cleaning of the remaining changes before doing the same for spellcheck.